go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -746,30 +748,31 @@ func (app *registryApplication) registerRuntime( // nolint: gocyclo
 
 	// Make sure that the entity or runtime has enough stake.
 	// Runtimes using the consensus layer governance model do not require stake.
-	if !params.DebugBypassStake && rt.GovernanceModel != registry.GovernanceConsensus {
+	if rtAddress := rt.StakingAddress(); !params.DebugBypassStake && rtAddress != nil {
 		claim := registry.StakeClaimForRuntime(rt.ID)
 		thresholds := registry.StakeThresholdsForRuntime(rt)
-		var acctAddr staking.Address
-		switch rt.GovernanceModel {
-		case registry.GovernanceEntity:
-			acctAddr = staking.NewAddress(rt.EntityID)
-		case registry.GovernanceRuntime:
-			acctAddr = ctx.CallerAddress()
-		default:
-			// Basic validation should have caught this, but just in case...
-			ctx.Logger().Debug("RegisterRuntime: invalid governance model")
-			return nil, registry.ErrInvalidArgument
-		}
 
-		if err = stakingState.AddStakeClaim(ctx, acctAddr, claim, thresholds); err != nil {
+		if err = stakingState.AddStakeClaim(ctx, *rtAddress, claim, thresholds); err != nil {
 			ctx.Logger().Debug("RegisterRuntime: insufficient stake",
 				"err", err,
 				"entity", rt.EntityID,
 				"runtime", rt.ID,
-				"account", acctAddr,
+				"account", rtAddress,
 			)
 			return nil, err
 		}
+
+		// If owner was updated remove stake claim from previous owner.
+		if existingRt != nil && *existingRt.StakingAddress() != *rtAddress {
+			if err = stakingState.RemoveStakeClaim(ctx, *existingRt.StakingAddress(), claim); err != nil {
```

Comment:
Given this is now doing multiple operations related to stake claims, it would be better to do the updates inside a transaction similar to how it is done in `registerNode`:
```golang
	// Start a new transaction and rollback in case we fail.
	ctx = ctx.NewTransaction()
	defer ctx.Close()

	// ...

	ctx.Commit()
```

Reply:
Added.